horace_core/sqw/@sqw/func_eval.m

```diff
@@ -116,10 +129,175 @@
 
     % If sqw object, fill every pixel with the value of its corresponding bin
     if sqw_type
-        s = replicate_array(wout(i).data.s, win(i).data.npix)';
-        wout(i).data.pix.signal = s;
-        wout(i).data.pix.variance = zeros(size(s));
-    elseif all_bins
-        wout(i).data.npix=ones(size(wout(i).data.npix));    % in this case, must set npix>0 to be plotted.
+        if ~opts.filebacked
+            s = repelem(wout(i).data.s(:), win(i).data.npix(:));
+            wout(i).data.pix.signal = s;
+            wout(i).data.pix.variance = zeros(size(s));
+        else
+            write_sqw_with_out_of_mem_pix(wout(i), opts.outfile{i});
+        end
+    elseif opts.all_bins
+        % in this case, must set npix>0 to be plotted
+        wout(i).data.npix=ones(size(wout(i).data.npix));
+    end
+
+    % Save to file if outfile argument is given
+    if ~isempty(opts.outfile) && ~isempty(opts.outfile{i}) && ~opts.filebacked
+        save(wout(i), opts.outfile{i});
+    end
+end  % end loop over input objects
+
+if opts.filebacked
+    % Return file names so we're not leaking file-backed objects
+    if numel(opts.outfile) > 1
+        wout = opts.outfile;
+    else
+        wout = opts.outfile{1};
+    end
+end
+
+end  % function
+
+
+% -----------------------------------------------------------------------------
+function [pars, opts] = parse_args(win, func_handle, pars, varargin)
+    if ~isa(func_handle, 'function_handle')
+        error( ...
+            'SQW:func_eval:invalid_argument', ...
+            'Argument ''func_handle'' must be a function handle.\nFound ''%s''.', ...
+            class(func_handle) ...
+        );
+    end
+
+    keyval_def = struct('all', false, 'outfile', '', 'filebacked', false);
+    flag_names = {'all'};
```

Comment:
This does look like it could be greatly simplified by `inputParser` having played with it a bit.